crates/red_knot/src/types/infer.rs

```diff
@@ -6,10 +6,8 @@ use crate::types::Type;
 use crate::FileId;
 use ruff_python_ast::AstNode;
 
-// TODO this should not take a &mut db, it should be a query, not a mutation. This means we'll need
-// to use interior mutability in TypeStore instead, and avoid races in populating the cache.
 #[tracing::instrument(level = "trace", skip(db))]
```

Comment:
Hmm. I was sure I added a FIXME here to figure out locking...

Reply:
You did. I pushed this on Sat morning, it wasn't rebased on your landed PRs yet. I rebased it now, and your comment is there.